### ISSUE TITLE
TK insertion depth

### ISSUE THREAD

**Issue by TEDSTANDAGE** [CCDevNelson/Snap-Updates#45]
The TKs have been reworked recently. Currently, when a TK is entered it defaults being inserted with a 1.5" inset from the depth of the left partition where the TK was placed, even if the right partition is shallower than the left partition, and it results in TKs being placed far in front of the section the TK lies under when that right partition is shallower than the left partition:

![image](https://user-images.githubusercontent.com/61025894/79998751-f0897d00-846f-11ea-93a7-57005be50a41.png)

Instead, if possible, the TK should be inset 1.5" from the shallowest partition depth (whether it's the left or right partition) rather than just defaulting to the left partition. Then, if the depth of either of the partitions the TK relates to is changed, the TK depth should adjust with it.

**Comment by TEDSTANDAGE** [comment]
If a TK covers more than one section and one of those sections changes depth while the other remains, the user is going to have to delete that TK assembly and re-insert separate TKs for the places it needs different TK depths, and that is okay.

**Comment by CCDevNelson** [comment]
This is exactly the opposite of what you want on the Top Shelf program. You still have the problem of not knowing what partitions are between your first and second pick. This one could take the smaller of the two if one is selected, but in this picture, wouldn't the 16" deep section have a 14.5" depth tow kick rather than an 10.5" toe kick under it anyways?

**Comment by TEDSTANDAGE** [comment]
That's all I'm asking for--taking the shallowest of the two you click. Also, the top shelf and the toe kick work the opposite in regard to whether you want to take the shallower or deeper of the partitions. Further, unlike topshelf, a single toekick assembly does not go under various section depths (whereas a continuous topshelf often goes over various depths sections, e.g., a 16" deep top shelf going over both a 12" section and a 16" section). 

In my example picture above, the left section correctly put a 10.5" TK under the 12" deep section because the left partition of that section is 12" deep. The middle section correctly put a 22.5" TK under the 24" deep section because the left partition of that section is 24" deep. However, the right section incorrectly put a 22.5" TK under the 16" deep section because the left partition of that section is 24" deep--however, if it had taken the shallower of the two I clicked in that section it would have put in a 14.5" TK because the right partition I clicked is 16" deep.

**Comment by CCDevNelson** [comment]
Yes, but that would then put a 14.5" section under the 24" opening, correct?

**Comment by TEDSTANDAGE** [comment]
No. There are 3 different depth sections in this example--12", 24", 16", from left to right. And there are three separate toe kick assemblies that I put in under each of those sections. I am not trying to span a single TK assembly across multiple sections of varying depths.  The 12" and 24" deep sections had TK depths placed correctly because the left partition depths matched the depth of those sections. The 16" deep section TK did not get placed correctly because the left partition of that section is 24" deep. The 24" deep would have the correct 22.5" TK depth because both the left and right partitions are 24" deep.

**Comment by TEDSTANDAGE** [comment]
I'll separate it out for clarity:

TK assembly correctly inserted as 10.5" deep for 12" deep section on left because left partition matches depth of section (under the rule I proposed it would work as well because it would take the shallower of the 12" deep left partition and the 24" deep right partition of the section):

![image](https://user-images.githubusercontent.com/61025894/80008718-f76abc80-847c-11ea-8d5d-4e800d81babe.png)

TK assembly corrected inserted as 22.5" deep for 24" deep section in middle because left partition matches depth of section (under the rule I proposed it would work as well because partitions of the section are 24" deep):

![image](https://user-images.githubusercontent.com/61025894/80008811-18331200-847d-11ea-9006-a69459017033.png)

TK assembly incorrectly inserted as 22.5" deep for 16" deep section on right because left partition is 24" deep unlike the depth of the section (under the rule I proposed, however, it would correctly put in a 14.5" deep TK assembly because it would take into account the shallower 16" deep right partition of the section):

![image](https://user-images.githubusercontent.com/61025894/80008913-31d45980-847d-11ea-8b40-9023756b3fe0.png)
